### DISCUSSION TITLE
Support the Sideberry extension

### DISCUSSION THREAD

**Discussion by scottwillmoore** [rafaelmardojai/firefox-gnome-theme#724] [Ideas]
It would be awesome if there was support for the [Sideberry extension](https://addons.mozilla.org/en-US/firefox/addon/sidebery/). I see that there is already support for the [Tab Center Reborn extension](https://addons.mozilla.org/en-US/firefox/addon/tabcenter-reborn/) in `configuration/extensions`. Sideberry is often [preferred](https://www.reddit.com/r/firefox/comments/uasl8d/tree_style_tab_vs_sidebery_vs_sidebar_tabs_vs/) as it provides a few additional features and is under active development.

To support it we should:

1. Provide an option to hide the horizontal tab bar.
2. Provide CSS for Sideberry to align it with the GNOME theme.

For (1), there is already some [documentation](https://www.reddit.com/r/firefox/comments/uasl8d/tree_style_tab_vs_sidebery_vs_sidebar_tabs_vs/) on how to hide the horizontal tab bar with this theme, but it would be great if this could be pulled upstream.

As for (2), which would be a bit more work, I would be happy to help. If so, do you have any resources on GNOME-like UI guidelines. I found the [HIG](https://developer.gnome.org/hig/index.html), but it doesn't appear to document the specific UI measurements.

**Comment by rafaelmardojai** [comment]
I'm moving this to a discussion since the README states:

> Be aware that extensions support are maintained by the community, so requests to support new extensions are not allowed and the included ones could get broken until someone shows up to fix them.

PRs are welcome of-course, but I don't want the issue tracker filled with extensions support requests.

**Comment by lionirdeadman** [comment]
I saw [this](https://github.com/mbnuqw/sidebery/wiki/Firefox-Styles-Snippets-(via-userChrome.css)#native-gnome-look-by-fletcher-alderton) which seems to do the work, would be neat to have it upstreamed if someone wants to do it.

**Comment by piousdeer** [comment]
Integrating Sidebery has actually been one of my toy projects that I've spent more time on than one would consider reasonable. It's pretty polished and employs some interesting techniques, including numerous userChrome.css tweaks, a Python script that generates CSS to replace built-in icons with Adwaita icons, and directly using libadwaita stylesheets in Sass.

https://github.com/rafaelmardojai/firefox-gnome-theme/assets/31318219/99407150-6d9b-4ac4-bbbe-f48f74325a73

(It seems the colors are somehow off in this recording 😕)

I'll link the project here when I release it.

**Reply by scottwillmoore** [reply]
Wow that looks amazing, I can't wait for it to be released. :smile:

**Comment by ghost** [comment]
in my case (maintainer of the [Tab Center Reborn extension](https://addons.mozilla.org/en-US/firefox/addon/tabcenter-reborn/) integration here, I actually stopped to add feature as a "native" vertical tab from Mozilla is in the work) and now i only fix bug and glitch due to update of firefox.

So I actually wait over an Mozilla support to skin it directly into this theme (if not already done by @rafaelmardojai) to make a last PR who will remove the Tab Center Reborn support to not have duplicate of feature (extension / native) that will complicate the maintenance for nothing.

**Reply by rafaelmardojai** [reply]
If I get to implement support for them will be a bit late, so any contributions when they get released would be welcome.
